prov/verbs/src/ep_rdm/verbs_rdm_cm.c

```diff
@@ -236,56 +236,38 @@ fi_ibv_rdm_tagged_init_qp_attributes(struct ibv_qp_init_attr *qp_attr,
 
 }
 
-static inline int
+static void
 fi_ibv_rdm_pack_cm_params(struct rdma_conn_param *cm_params,
-			  struct fi_ibv_rdm_conn *conn,
-			  struct fi_ibv_rdm_ep *ep)
+			  struct fi_conn_priv_params *priv,
+			  const struct fi_ibv_rdm_conn *conn,
+			  const struct fi_ibv_rdm_ep *ep)
 {
-	char *p;
-
 	memset(cm_params, 0, sizeof(struct rdma_conn_param));
 	cm_params->responder_resources = 2;
 	cm_params->initiator_depth = 2;
+	cm_params->private_data = priv;
 
-	cm_params->private_data_len = FI_IBV_RDM_DFLT_ADDRLEN;
-
-	if ((conn->cm_role != FI_VERBS_CM_SELF) && (conn->r_mr && conn->s_mr)) {
-		cm_params->private_data_len += sizeof(conn->r_mr->rkey);
-		cm_params->private_data_len += sizeof(conn->remote_rbuf_mem_reg);
-		cm_params->private_data_len += sizeof(conn->s_mr->rkey);
-		cm_params->private_data_len += sizeof(conn->remote_sbuf_mem_reg);
-	}
-
-	cm_params->private_data = calloc(1, cm_params->private_data_len);
-	if (!cm_params->private_data)
-		return -FI_ENOMEM;
-
-	p = (char *) cm_params->private_data;
-	memcpy(p, &ep->my_addr, FI_IBV_RDM_DFLT_ADDRLEN);
-	p += FI_IBV_RDM_DFLT_ADDRLEN;
+	memcpy(priv->addr, &ep->my_addr, FI_IBV_RDM_DFLT_ADDRLEN);
 
-	if ((conn->cm_role != FI_VERBS_CM_SELF) && (conn->r_mr && conn->s_mr)) {
-		memcpy(p, &conn->r_mr->rkey, sizeof(conn->r_mr->rkey));
-		p += sizeof(conn->r_mr->rkey);
-		memcpy(p, &conn->rbuf_mem_reg, sizeof(conn->rbuf_mem_reg));
-		p += sizeof(conn->rbuf_mem_reg);
+	if ((conn->cm_role != FI_VERBS_CM_SELF) && conn->r_mr && conn->s_mr) {
+		cm_params->private_data_len = sizeof(*priv);
 
-		memcpy(p, &conn->s_mr->rkey, sizeof(conn->s_mr->rkey));
-		p += sizeof(conn->s_mr->rkey);
-		memcpy(p, &conn->sbuf_mem_reg, sizeof(conn->sbuf_mem_reg));
-		p += sizeof(conn->sbuf_mem_reg);
+		priv->rbuf_rkey = conn->r_mr->rkey;
+		priv->rbuf_mem_reg = conn->rbuf_mem_reg;
+		priv->sbuf_rkey = conn->s_mr->rkey;
+		priv->sbuf_mem_reg = conn->sbuf_mem_reg;
+	} else {
+		cm_params->private_data_len = FI_IBV_RDM_DFLT_ADDRLEN;
```

Comment:
We don't need a private data for the `cm_role == FI_VERBS_CM_SELF` case.
Can we just do `cm_params->private_data = NULL`  in this case?
If you have no time for that, I'm fine with prepared patch and I could address this fix in the new PR.

Reply:
That's a change with the previous code, so that should be in a different patch.

Reply:
I can make a patch for that as well.

Reply:
> That's a change with the previous code, so that should be in a different patch.

Yes, I agree.

> I can make a patch for that as well.

If possible, it would be great. But you're doing a lot of useful work for the verbs provider. I don't want to burden you with verbs related work and can make a patch by myself.

Thanks for your finding and prepared improvements!